core/trino-main/src/main/java/io/trino/sql/rewrite/ShowQueriesRewrite.java

```diff
@@ -564,12 +566,21 @@ private Query showCreateMaterializedView(ShowCreate node)
                     false,
                     viewDefinition.get().getGracePeriod()
                             .map(DateTimeUtils::formatDayTimeInterval),
-                    Optional.empty(), // TODO support WHEN STALE
+                    viewDefinition.get().getWhenStaleBehavior()
+                            .map(Visitor::toSqlWhenStaleBehavior),
```

Comment:
given INLINE is the default, we could `viewDefinition.get().getWhenStaleBehavior().orElse(INLINE)`